clients/js/src/index.ts

```diff
@@ -204,11 +240,51 @@ export class Collection {
 
   }
 
+  public async update(
+    ids: string | string[],
+    embeddings?: number[] | number[][],
+    metadatas?: object | object[],
+    documents?: string | string[],
+  ) {
+    if ((embeddings === undefined) && (documents === undefined) && (metadatas === undefined)) {
+      throw new Error(
+        "embeddings, documents, and metadatas cannot all be undefined",
+      );
+    } else if ((embeddings === undefined) && (documents !== undefined)) {
+      const documentsArray = toArray(documents);
+      if (this.embeddingFunction !== undefined) {
+        embeddings = await this.embeddingFunction.generate(documentsArray)
+      } else {
+        throw new Error(
+          "embeddingFunction is undefined. Please configure an embedding function",
+        );
+      }
+    }
+
+    var resp = await this.api.update({
+      collectionName: this.name,
+      updateEmbedding: {
+        ids: toArray(ids),
+        embeddings: (embeddings ? toArrayOfArrays(embeddings) : undefined),
+        documents: toArray(documents),
+        metadatas: toArray(metadatas),
+      },
+    }).then(function (response) {
+      return response.data;
+    }).catch(function ({ response }) {
+      return response.data;
+    });
+
+    return resp
+  }
+
   public async query(
     query_embeddings: number[] | number[][] | undefined,
     n_results: number = 10,
     where?: object,
-    query_text?: string | string[],
+    query_text?: string | string[], // TODO: should be named query_texts to match python API
```

Comment:
Should we complete this TODO?

Reply:
this would cause a breaking change in the API, and I was trying to constrain this diff to have no breaking changes. 

I think we will soon move over to object-based inputs, eg as laid out here https://github.com/chroma-core/chroma/issues/254, and i was planning on making that change then. 

thoughts?